src/display/fetch_stream.js

```diff
@@ -121,6 +124,8 @@ class PDFFetchStreamReader {
       createFetchOptions(headers, this._withCredentials, this._abortController)
     )
       .then(response => {
+        stream._responseOrigin.resolve(getResponseOrigin(response.url));
```

Comment:
This part may not be reached when the network request fails. Is it desirable to indefinitely postpone the resolution of the other responses?

Reply:
The latest version of the patch updates the unit-tests to agree with the "actual" usage, since that removes the need for the additional Promises in the code.